examples/texture-arrays/shader.frag

```diff
@@ -8,11 +10,5 @@ layout(set = 0, binding = 0) uniform texture2D u_Textures[2];
 layout(set = 0, binding = 1) uniform sampler u_Sampler;
 
 void main() {
-    if (v_Index == 0) {
-        o_Color = vec4(texture(sampler2D(u_Textures[0], u_Sampler), v_TexCoord).rgb, 1.0);
-    } else if (v_Index == 1) {
-        o_Color = vec4(texture(sampler2D(u_Textures[1], u_Sampler), v_TexCoord).rgb, 1.0);
-    } else {
-        o_Color = vec4(0.0, 0.0, 1.0, 1.0);
-    }
+    o_Color = vec4(texture(sampler2D(u_Textures[v_Index], u_Sampler), v_TexCoord).rgb, 1.0);
```

Comment:
How do you feel about making this example to support both uniform and non-uniform indexing?
That could be a good demonstration of how to support optional extensions. We'd just have 2 different shaders here, one picked at initialization.

Reply:
Sure